lib/segment/src/common/mmap_ops.rs

```diff
@@ -0,0 +1,62 @@
+use std::fs::OpenOptions;
+use std::mem;
+use std::mem::size_of;
+use std::path::Path;
+
+use memmap2::{Mmap, MmapMut, MmapOptions};
+
+use crate::entry::entry_point::OperationResult;
+use crate::madvise;
+
+pub fn open_read_mmap(path: &Path) -> OperationResult<Mmap> {
+    let file = OpenOptions::new()
+        .read(true)
+        .write(false)
+        .append(true)
+        .create(true)
+        .open(path)?;
+
+    let mmap = unsafe { MmapOptions::new().map(&file)? };
+    madvise::madvise(&mmap, madvise::get_global())?;
+    Ok(mmap)
+}
+
+pub fn open_write_mmap(path: &Path) -> OperationResult<MmapMut> {
+    let file = OpenOptions::new()
+        .read(true)
+        .write(true)
+        .create(false)
+        .open(path)?;
+
+    let mmap = unsafe { MmapMut::map_mut(&file)? };
+    madvise::madvise(&mmap, madvise::get_global())?;
+    Ok(mmap)
+}
+
+pub fn transmute_from_u8<T>(data: &[u8]) -> &T {
+    debug_assert!(data.len() == size_of::<T>());
+    let ptr = data.as_ptr() as *const T;
+    unsafe { &*ptr }
+}
+
+pub fn transmute_to_u8<T>(v: &T) -> &[u8] {
+    unsafe { std::slice::from_raw_parts(v as *const T as *const u8, mem::size_of_val(v)) }
+}
+
+pub fn transmute_from_u8_to_array<T>(data: &[u8]) -> &[T] {
```

Comment:
Nitpick: this is not an [`array`](https://doc.rust-lang.org/std/primitive.array.html) but a [`slice`](https://doc.rust-lang.org/std/primitive.slice.html). Same goes for the two functions below.

Reply:
:+1: